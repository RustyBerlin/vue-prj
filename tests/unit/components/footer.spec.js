import { mount, RouterLinkStub, createLocalVue } from '@vue/test-utils';
import Component from '../../../components/Footer.vue';

describe('Test', () => {
  let wrapper;
  const localVue = createLocalVue();
  localVue.component('router-link', RouterLinkStub);

  beforeEach(() => {
    wrapper = mount(Component, {
      localVue,
    });
  });

  test('snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('check html elements', () => {
    const htmlElements = [
      'footer',
      'ul',
      'li',
      'a',
    ]

    htmlElements.forEach(element => {
      expect(wrapper.contains(element)).toBe(true);
    });
  });
});
