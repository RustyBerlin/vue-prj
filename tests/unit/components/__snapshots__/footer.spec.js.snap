// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot 1`] = `"<footer><ul class=\\"module\\"><li><a>Home</a></li> <li><a>Imprint</a></li> <li><a href=\\"#top\\">Top</a></li></ul></footer>"`;
